### PR TITLE
feat: Download E2EI certificate

### PR DESCRIPTION
**Pull request by przemvs**
## Description
Downloading certificate based on string which we get.

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.test.tsx [hunk 1]
```diff
@@ -33,18 +33,19 @@ Object.assign(navigator, {
...
     const {getByTestId} = render(<CertificateDetailsModal {...defaultProps} />);
 
-    const downloadButton = getByTestId('download-certificate');
+    const downloadButton = getByTestId('download-certificate') as HTMLButtonElement;
     expect(downloadButton).toBeDefined();
+    expect(downloadButton.disabled).toBe(false);
+
     fireEvent.click(downloadButton);
 
-    expect(defaultProps.downloadCertificate).toHaveBeenCalled();
```
Comment (atomrc) [inline src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.test.tsx:47]: why removing this expectation? It seems like a valid one

Comment (atomrc) [reply src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.test.tsx:47]: Ah, nevermind, the callback has been removed

File: src/script/page/MainContent/panels/preferences/devices/components/E2EICertificateDetails/E2EICertificateDetails.tsx [hunk 2]
```diff
@@ -129,11 +129,7 @@ export const E2EICertificateDetails = ({core = container.resolve(Core), isMLSVer
         )}
 
         {isCertificateDetailsModalOpen && (
-          <CertificateDetailsModal
-            certificate={certificate}
-            onClose={() => setIsCertificateDetailsModalOpen(false)}
-            downloadCertificate={getCertificate}
```
Comment (atomrc) [inline src/script/page/MainContent/panels/preferences/devices/components/E2EICertificateDetails/E2EICertificateDetails.tsx:132]: The `getCerfiticate` can probably be removed, right?

**Review by atomrc — APPROVED** [review]

File: src/script/page/MainContent/panels/preferences/devices/components/E2EICertificateDetails/E2EICertificateDetails.tsx (hunk 2 above)
Comment (przemvs) [reply src/script/page/MainContent/panels/preferences/devices/components/E2EICertificateDetails/E2EICertificateDetails.tsx:132]: This is a different function probably :D On design we have Get Certificate and Download Certificate, so we getting certificate by enrolling if i'm good thinking. Get certificate will be preapred there: https://github.com/wireapp/wire-webapp/pull/15677

File: src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.tsx [hunk 3]
```diff
@@ -26,24 +26,36 @@ import {t} from 'Util/LocalizerUtil';
...
+export const CertificateDetailsModal = ({certificate, onClose}: CertificateDetailsModalProps) => {
   const [isTextCopied, setIsTextCopied] = useState(false);
   const [isDownloading, setIsDownloading] = useState(false);
 
   const onDownload = () => {
     setIsDownloading(true);
-    downloadCertificate();
 
-    setTimeout(() => {
+    const downloadLink = document.createElement('a');
```
Comment (PatrykBuniX) [inline src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.tsx]: suggestion: There's `downloadFile` function in `util.ts` file, maybe we could use it (or extend it to match this use case if it's missing something).

Comment (przemvs) [reply src/script/components/Modals/CertificateDetailsModal/CertificateDetailsModal.tsx]: https://github.com/wireapp/wire-webapp/pull/15794/commits/563f26aac62f3be734a935a32c6930f1e22084c4

**Review by arjita-mitra — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]
Nice 💪

**Pull request merged**